### PR TITLE
Add effective stress materials and auxkernel

### PR DESCRIPTION
**Pull request by arovinelli**
This pull request adds to Deer the capability of calculating effective stress as material property, both in bulk and over an interface. There is also an aux kernel doing the same job.

### CODE REVIEW

File: src/auxkernels/EffectiveStressAux.C [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+//* All rights reserved, see COPYRIGHT for full restrictions
+//* https://github.com/idaholab/moose/blob/master/COPYRIGHT
+//*
+//* Licensed under LGPL 2.1, please see LICENSE for details
+//* https://www.gnu.org/licenses/lgpl-2.1.html
+
+#include "EffectiveStressAux.h"
+#include "EffectiveStressTools.h"
+
+registerMooseObject("DeerApp", EffectiveStressAux);
```
Comment (reverendbedford) [inline src/auxkernels/EffectiveStressAux.C]: Capitalization, as per usual.

File: include/auxkernels/EffectiveStressAux.h [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+
+  /**
+   * Determines the information to be extracted from the tensor by using the
+   * EffectiveStressTools namespace, e.g., vonMisesStressL2norm, MaxPrincipal
+   * eigenvalue, etc.
+   */
+  MooseEnum _scalar_type;
+  std::vector<Real> _params_vector;
+
+  const Point _point1;
```
Comment (reverendbedford) [inline include/auxkernels/EffectiveStressAux.h]: Are these ever actually used?  It doesn't look like it, so they probably should be gone.

File: src/auxkernels/EffectiveStressAux.C [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+      _point2(parameters.get<Point>("point2")),
+      _input_direction(parameters.get<Point>("direction") /
+                       parameters.get<Point>("direction").norm()) {
+
+  if (_scalar_type == 2 && _params_vector.size() != 1)
+    mooseError("the huddleston effective stress requires the parameters b to "
+               "be defined");
+  if (_scalar_type == 3) {
+    if (_params_vector.size() != 2)
+      mooseError("the hayhurs effective stress requires the parameters alpha, "
```
Comment (reverendbedford) [inline src/auxkernels/EffectiveStressAux.C]: Spelling, capitalization.

File: src/auxkernels/EffectiveStressAux.C [hunk 4]
```diff
@@ -0,0 +1,69 @@
...
+      _tensor(getMaterialProperty<RankTwoTensor>("stress")),
+      _scalar_type(getParam<MooseEnum>("effective_stress_type")),
+      _params_vector(getParam<std::vector<Real>>("params_vector")),
+      _point1(parameters.get<Point>("point1")),
+      _point2(parameters.get<Point>("point2")),
+      _input_direction(parameters.get<Point>("direction") /
+                       parameters.get<Point>("direction").norm()) {
+
+  if (_scalar_type == 2 && _params_vector.size() != 1)
+    mooseError("the huddleston effective stress requires the parameters b to "
```
Comment (reverendbedford) [inline src/auxkernels/EffectiveStressAux.C]: Capitalization.

File: src/auxkernels/EffectiveStressAux.C [hunk 5]
```diff
@@ -0,0 +1,69 @@
...
+               "be defined");
+  if (_scalar_type == 3) {
+    if (_params_vector.size() != 2)
+      mooseError("the hayhurs effective stress requires the parameters alpha, "
+                 "beta and gamma to be defined");
+    _params_vector.push_back(1. - _params_vector[0] - _params_vector[1]);
+  }
+  if (_scalar_type == 6 && _params_vector.size() != 1)
+    mooseError(
+        "the RCCMRXMises effective stress requires the parameters alpha to "
```
Comment (reverendbedford) [inline src/auxkernels/EffectiveStressAux.C]: See above.

File: src/auxkernels/EffectiveStressAux.C [hunk 6]
```diff
@@ -0,0 +1,69 @@
...
+        "be defined");
+  if (_scalar_type == 7 && _params_vector.size() != 1)
+    mooseError(
+        "the RCCMRXTresca effective stress requires the parameters alpha to "
```
Comment (reverendbedford) [inline src/auxkernels/EffectiveStressAux.C]: See above.

File: src/materials/EffectiveStressInterfaceMaterial.C [hunk 7]
```diff
@@ -0,0 +1,93 @@
...
+//* Licensed under LGPL 2.1, please see LICENSE for details
+//* https://www.gnu.org/licenses/lgpl-2.1.html
+
+#include "EffectiveStressInterfaceMaterial.h"
+#include "EffectiveStressTools.h"
+#include "InterfaceValueTools.h"
+
+registerMooseObject("DeerApp", EffectiveStressInterfaceMaterial);
+
+template <> InputParameters validParams<EffectiveStressInterfaceMaterial>() {
```
Comment (reverendbedford) [inline src/materials/EffectiveStressInterfaceMaterial.C]: Capitalization and spelling.

File: src/materials/EffectiveStressInterfaceMaterial.C [hunk 8]
```diff
@@ -0,0 +1,93 @@
...
+      _effective_stress_type(getParam<MooseEnum>("effective_stress_type")),
+      _interface_value_type(parameters.get<MooseEnum>("interface_value_type")),
+      _params_vector(getParam<std::vector<Real>>("params_vector")),
+      _point1(parameters.get<Point>("point1")),
+      _point2(parameters.get<Point>("point2")),
+      _input_direction(parameters.get<Point>("direction") /
+                       parameters.get<Point>("direction").norm()),
+      _stateful(getParam<bool>("stateful")) {
+  if (_effective_stress_type == 2 && _params_vector.size() != 1)
+    mooseError("the huddleston effective stress requires the parameters b to "
```
Comment (reverendbedford) [inline src/materials/EffectiveStressInterfaceMaterial.C]: The usual comments on capitalization and spelling. (and continued below, i.e. hayhurs = Hayhurst).

File: src/materials/EffectiveStressMaterial.C [hunk 9]
```diff
@@ -0,0 +1,82 @@
...
+//* Licensed under LGPL 2.1, please see LICENSE for details
+//* https://www.gnu.org/licenses/lgpl-2.1.html
+
+#include "EffectiveStressMaterial.h"
+#include "EffectiveStressTools.h"
+#include "InterfaceValueTools.h"
+
+registerMooseObject("DeerApp", EffectiveStressMaterial);
+
+template <> InputParameters validParams<EffectiveStressMaterial>() {
```
Comment (reverendbedford) [inline src/materials/EffectiveStressMaterial.C]: Capitalization and spelling...

File: src/materials/EffectiveStressMaterial.C [hunk 10]
```diff
@@ -0,0 +1,82 @@
...
+  params.addParam<Point>("point1", Point(0, 0, 0),
+                         "Start point for axis used to calculate some "
+                         "cylindrical material tensor quantities");
+  params.addRequiredParam<MaterialPropertyName>(
+      "effective_stress_mp_name", "the name of the new  material_property");
+  params.addParam<Point>(
+      "point2", Point(0, 1, 0),
+      "End point for axis used to calculate some material tensor quantities");
+  params.addParam<Point>("direction", Point(0, 0, 1), "Direction vector");
+  params.addParam<bool>(
```
Comment (reverendbedford) [inline src/materials/EffectiveStressMaterial.C]: Can you take this decision out of their hands?

File: src/materials/EffectiveStressMaterial.C [hunk 11]
```diff
@@ -0,0 +1,82 @@
...
+      _point2(parameters.get<Point>("point2")),
+      _input_direction(parameters.get<Point>("direction") /
+                       parameters.get<Point>("direction").norm()),
+      _stateful(getParam<bool>("stateful")) {
+  if (_effective_stress_type == 2 && _params_vector.size() != 1)
+    mooseError("the huddleston effective stress requires the parameters b to "
+               "be defined");
+  if (_effective_stress_type == 3) {
+    if (_params_vector.size() != 2)
+      mooseError("the hayhurs effective stress requires the parameters alpha, "
```
Comment (reverendbedford) [inline src/materials/EffectiveStressMaterial.C]: ...

File: src/materials/EffectiveStressMaterial.C (hunk 10 above)
Comment (arovinelli) [reply src/materials/EffectiveStressMaterial.C]: if you are taking about making the material property stateful moose should do it automatically if requested somewhere else, but apparently it doesn't work in this case.

**Comment by arovinelli** [comment]
>     1. The effective_stress_interface_material test doesn't work.  This time you have the gold file, but the gold file and the output produced by your .i file don't have the right filenames.

fixed the name
 
>     2. Lots of spelling errors in various places.

my Achille's heel should be fixed

>     3. I have no idea what the point/direction stuff does.  Is it even used?

these parameters are required from RwankTwoScalarUtils. I managed to move them in the EffectiveStressUtils file so we will not see them anymore

**Pull request merged**